daemon/commit.go

```diff
@@ -20,7 +22,18 @@ type ContainerCommitConfig struct {
 // The image can optionally be tagged into a repository.
 func (daemon *Daemon) Commit(container *Container, c *ContainerCommitConfig) (*image.Image, error) {
 	if c.Pause && !container.isPaused() {
-		container.pause()
+		if err := container.pause(); err != nil {
+			switch err.(type) {
+			case errcode.Error:
+				daError, _ := err.(errcode.Error)
+				if daError.ErrorCode() != derr.ErrorCodeNotRunning && daError.ErrorCode() != derr.ErrorCodeAlreadyPaused {
```

Comment:
I don't think `ErrorCodeNotRunning` and `ErrorCodeAlreadyPaused` are the same. We cannot unpause it in the defer below if the container is not running because it's stopped.

I think we should only skip returning here when the error is `ErrCodeAlreadyPaused`, and we should only try to unpause when it was really paused.